### PR TITLE
Hooks wrapper mod_*

### PR DESCRIPTION
**Pull request by gustawlippa**
This PR changes not grouped `mod_*` modules to use the new `mongoose_hooks` module.

### CODE REVIEW

File: src/mod_commands.erl [hunk 1]
```diff
@@ -261,12 +261,17 @@ unregister(Host, User) ->
...
-    Packet = build_packet(message_chat, Body),
     F = jid:from_binary(From),
     T = jid:from_binary(To),
-    ejabberd_hooks:run(user_send_packet,
-                       F#jid.lserver,
-                       [F, T, Packet]),
+    Packet = build_message(From, To, Body),
+    Acc0 = mongoose_acc:new(#{location => ?LOCATION,
+                                       lserver => F#jid.lserver,
+                                       element => Packet}),
```
Comment (chrzaszcz) [inline src/mod_commands.erl]: Fix indentation

File: src/mod_ping.erl [hunk 2]
```diff
@@ -176,8 +176,8 @@ handle_info({timeout, _TRef, {ping, JID}},
     T0 = erlang:monotonic_time(millisecond),
     F = fun(_From, _To, Acc, Response) ->
                 TDelta = erlang:monotonic_time(millisecond) - T0,
-                NewAcc = ejabberd_hooks:run_fold(user_ping_response, State#state.host,
-                                                 Acc, [JID, Response, TDelta]),
+                NewAcc = mongoose_hooks:user_ping_response(State#state.host,
+                                                 Acc, JID, Response, TDelta),
```
Comment (chrzaszcz) [inline src/mod_ping.erl]: Fix indentation

File: src/mod_vcard.erl [hunk 3]
```diff
@@ -497,8 +497,8 @@ do_route(_VHost, From, To, Acc, #iq{type = set,
 do_route(VHost, From, To, _Acc, #iq{type = get,
                                        xmlns = ?NS_DISCO_INFO,
                                        lang = Lang} = IQ) ->
-    Info = ejabberd_hooks:run_fold(disco_info, VHost, [],
-                                   [VHost, ?MODULE, <<"">>, <<"">>]),
+    Info = mongoose_hooks:disco_info(VHost, [],
+                                     ?MODULE, <<"">>, <<"">>),
```
Comment (chrzaszcz) [inline src/mod_vcard.erl]: Minor: maybe fit in one line?

File: src/mongoose_hooks.erl [hunk 4]
```diff
@@ -131,6 +138,26 @@
          pubsub_delete_node/5,
          pubsub_publish_item/7]).
 
+-spec adhoc_local_items(LServer, Acc, From, To, Lang) -> Result when
```
Comment (chrzaszcz) [inline src/mongoose_hooks.erl]: No docs for these functions, maybe add them later

Comment (gustawlippa) [reply src/mongoose_hooks.erl]: True, but I am not sure how adhoc works and what exactly is happening here. I didn't want to spend time investigating to deeply, especially seeing that this hook is not covered by tests.

**Pull request merged**